cmake/external/abseil-cpp.cmake

```diff
@@ -13,13 +13,26 @@ else()
   set(ABSL_PATCH_COMMAND git apply --ignore-space-change --ignore-whitespace ${PROJECT_SOURCE_DIR}/patches/abseil/Fix_Nvidia_Build_Break.patch)
 endif()
 
-FetchContent_Declare(
-    abseil_cpp
-    PREFIX "${CMAKE_CURRENT_BINARY_DIR}/abseil-cpp"
-    BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/abseil-cpp"
-    URL https://github.com/abseil/abseil-cpp/archive/refs/tags/20211102.0.zip
-    PATCH_COMMAND ${ABSL_PATCH_COMMAND}
-)
+# CMake prior to 3.18.2 may over-patch during rebuild.
+# https://gitlab.kitware.com/cmake/cmake/-/issues/21086
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.18.2)
+  FetchContent_Declare(
+      abseil_cpp
+      PREFIX "${CMAKE_CURRENT_BINARY_DIR}/abseil-cpp"
+      BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/abseil-cpp"
+      GIT_REPOSITORY https://github.com/abseil/abseil-cpp.git
+      GIT_TAG 20211102.0
+      PATCH_COMMAND ${ABSL_PATCH_COMMAND}
```

Comment:
>PATCH_COMMAN [](http://example.com/codeflow?start=6&length=12)

Last time I tried this a repeated local dev build would fail because patch command would fail and that would fail the entire build. Is this now fixed?

Reply:
1. Could you be more specific about the version, including the 3rd part? CMake do have backported patches. If it's related we may need a few more conditions for those versions.
2. Which generator are you using? When using ninja on Linux it seems fine.

Reply:
They claimed it's fixed by:
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5136
<img width="1724" alt="image" src="https://user-images.githubusercontent.com/5203025/164770752-ee6693b4-7c57-4a1f-b63b-12fb9d5d0f45.png">